src/loader/loader.ts

```diff
@@ -21,14 +21,52 @@ interface PendingExtensionWorker {
     reject: (value: unknown) => void;
 }
 
+interface ChibiCompatibleScript extends HTMLScriptElement {
+    Scratch?: Context;
+}
+
 export interface ScratchExtension {
     id: string;
     url: string;
     env: 'sandboxed' | 'unsandboxed';
     info: ExtensionMetadata;
     instance: string | ExtensionClass; // The serviceName or extensionClass.
 }
-
+/**
+ * Load unsandboxed extensions with ESM support.
+ * @param extensionURL Extension's URL.
+ * @param ctx Scratch context.
+ * @returns A promise to wait for the extensions to be loaded.
+ */
+async function loadChibiExtension (extensionURL: string, ctx: Context): Promise<void> {
+    const resp = await fetch(extensionURL, {
+        cache: 'no-cache'
+    });
+    const code = await resp.text();
+    return new Promise<void>((resolve, reject) => {
+        const elem = document.createElement('script') as ChibiCompatibleScript;
+        elem.id = 'chibiExtension';
```

Comment:
Will multiple elements with the same id cause problems?
Consider loading multiple non-sandbox extensions at once, which may cause problems